### PR TITLE
Generate random realizations of P1D measurements according to cov matrix

### PR DESCRIPTION
**Pull request by jchavesmontero**
This PR addresses #24.

### CODE REVIEW

**Comment by andreufont** [comment]
Let's merge #54 first, update this and review. 

However, I would like to chat about this in person (are you at IFAE?), since this is not quite what I had imagined.

File: src/cup1d/data/data_fid_eBOSS_mock.py
Comment (andreufont) [inline src/cup1d/data/data_fid_eBOSS_mock.py:1]: You should delete this file, isn't it?

**Comment by andreufont** [comment]
Let's have the chat here. I was imagining that the option to add noise would be passed in the construction of the P1D measurement, and that the perturbed P1D would be stored as if it was a normal P1D. That way you make sure that any function that interacts with the P1D object will always get the perturbed P1D, for plotting, for the likelihood, etc.

I thought I would add this to one of the mock_p1d objects, and not in the base_p1d class, but I guess it doesn't hurt to have it there, and this way you can recycle the code for the different p1d objects that are mock data. The proper "object-oriented" way to do this would be to have a base_mock_p1d class that inherits base_p1d and adds this functionality, and then objects like eboss_mock or the older mock objects would inherit base_mock_p1d, but I have been known in the past for going too deep into object-orienting programing, so I will not insist :-)

File: src/cup1d/data/data_fid_eBOSS_mock.py
Comment (andreufont) [reply src/cup1d/data/data_fid_eBOSS_mock.py:1]: @jchavesmontero - I thought you had already deleted this file?

**Review by andreufont — APPROVED** [review]
@jchavesmontero - this looks great! I added two minor comments (one file can be deleted, and one argument is not necessary), but happy to merge otherwise. 

We should modify a couple of other objects (`data_gadget`, `data_nyx`, `data_mock`) so that they also inherit `base_p1d_mock` instead of `base_p1d_data`. This would allow us to do noisy mocks of these as well.

File: src/cup1d/data/data_fid_eBOSS_mock.py
Comment (jchavesmontero) [reply src/cup1d/data/data_fid_eBOSS_mock.py:1]: I deleted **locally** a couple of times, now I did it in github, sorry

File: src/cup1d/data/base_p1d_mock.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        if add_noise:
+            Pk_perturb_kms = self.get_Pk_iz_perturbed(
+                Pk_kms, cov_Pk_kms, seed=seed
+            )
+        else:
+            Pk_perturb_kms = Pk_kms
+
+        super().__init__(z, k_kms, Pk_perturb_kms, cov_Pk_kms)
+
+    def get_Pk_iz_perturbed(self, Pk_kms, cov_Pk_kms, nsamples=1, seed=0):
```
[the remark this answers is not in the document]
Comment (jchavesmontero) [reply src/cup1d/data/base_p1d_mock.py:23]: It is useful to check that we recover the expected median and std when computing the average and std of many realizations. I was using this functionality in the notebook. On the other hand, I do not expect to use this for anything else, and you may have noticed that the predetermined option is nsamples=1

**Comment by jchavesmontero** [comment]
> @jchavesmontero - this looks great! I added two minor comments (one file can be deleted, and one argument is not necessary), but happy to merge otherwise.
> 
> We should modify a couple of other objects (`data_gadget`, `data_nyx`, `data_mock`) so that they also inherit `base_p1d_mock` instead of `base_p1d_data`. This would allow us to do noisy mocks of these as well.

Sure! I just wanted to be sure that everything was looking good before doing so. Working on this now

**Pull request merged**